src/test/scala/mesosphere/marathon/MarathonSchedulerActorTest.scala

```diff
@@ -84,6 +86,33 @@ class MarathonSchedulerActorTest extends MarathonActorSupport
     }
   }
 
+  test("Terminal tasks should not be submitted in reconciliation") {
+    val f = new Fixture
+    import f._
+    val app = AppDefinition(id = "/test-app".toPath, instances = 1)
+    val instance = TestInstanceBuilder.newBuilder(app.id).addTaskUnreachable(containerName = Some("unreachable")).addTaskRunning().addTaskGone(containerName = Some("gone")).getInstance()
```

Comment:
Can you please add _all_ available `Terminal` status to the instance and verify explicitly that none of these is sent for reconciliation? I'd like to see
> Created
> Error
> Failed
> Finished
> Killed
> Killing
> Running
> Staging
> Starting
> Unreachable
> Gone
> Dropped
> Unknown

and a verify that status is only sent for

> Created
> Killing
> Running
> Staging
> Starting
> Unreachable

Btw: It would be nice to extract the underlying functionality _which_ task status are sent to mesos. Out of scope for this PR though.